sql/internal/specutil/convert.go

```diff
@@ -71,6 +73,14 @@ type (
 		Func  func(*schema.Func) (*sqlspec.Func, error)
 		Proc  func(*schema.Proc) (*sqlspec.Func, error)
 	}
+	// RefNamer is an interface for objects that can
+	// return their reference and type name.
+	RefNamer interface {
+		// Ref returns the reference to the object.
+		Ref() *schemahcl.Ref
+		// TypeName returns the type and name of the object.
+		TypeName() (string, string)
```

Comment:
I think better to be explicit that these are spec and type names. So either SpecTypeName, or SpecType+SpecName, or return specutil.TypeName instead.